src/Sentry/Scope.cs

```diff
@@ -184,7 +184,7 @@ public User User
         /// <inheritdoc />
         public IReadOnlyDictionary<string, string> Tags => _tags;
 
-        private readonly ConcurrentBag<Attachment> _attachments = new();
+        private ConcurrentBag<Attachment> _attachments = new();
```

Comment:
Can you guard the `readonly` through:

the same `#ifdef` you use below? We should keep readonly if it can be so on newer targets.